tests/unit/dump.tcl

```diff
@@ -124,6 +124,25 @@ start_server {tags {"dump"}} {
         close_replication_stream $repl
     } {} {needs:repl}
 
+    test {RESTORE key with future RDB version, strict version check} {
+        #              str len    RDB 222 CRC64 checksum
+        #               |   |      |       |
+        set bar_dump "\x00\x03bar\xde\x00\x0fYUza\xd3\xec\xe0"
+        catch {r restore foo 0 $bar_dump replace} e
+        set e
+    } {ERR DUMP payload version or checksum are wrong}
```

Comment:
```suggestion
       assert_error {ERR DUMP payload version or checksum are wrong} {r restore foo 0 $bar_dump replace}
    }
```
Minor thing, it seems silly to have so many lines when we have a helper for this.